### PR TITLE
Refactor `take_all_your`

### PR DESCRIPTION
**Pull request by JohnAHolt**

### CODE REVIEW

**Comment by JohnAHolt** [comment]
@simonelnahas

File: lib/go_fish/player.ex [hunk 1]
```diff
@@ -140,4 +125,30 @@ defmodule GoFish.Player do
   #     {:give-me-all-your, taker, asking-value} ->
   #       IO.puts("not implemented") #TODO
   # end
+
+
+
+
+  # Helper functions
+
+  def go_fish(state) do
```
Comment (simonelnahas) [inline lib/go_fish/player.ex]: My personal preference is to always have the function defined before it is used. But I know not everyone would agree with this.

File: lib/go_fish/player.ex [hunk 2]
```diff
@@ -140,4 +125,30 @@ defmodule GoFish.Player do
...
+    IO.puts("They didn't have the requested value so I go fishing")
+    case GoFish.Ocean.take_card() do
+        {:card, card} ->
+          IO.puts("I drew the card #{inspect(card)} from the ocean")
+          newState = %{state |
+                        :isMyTurn => false,
+                        :hand => [card | Map.get(state, :hand)]}
+          {:reply, :went_fishing, newState}
```
Comment (simonelnahas) [inline lib/go_fish/player.ex]: You could refactor this into a function as well

**Review by simonelnahas — APPROVED** [review]
LGTM

**Pull request merged**